packages/ui/src/molecules/RadioGroup/RadioOption.tsx

```diff
@@ -0,0 +1,49 @@
+import type { InputHTMLAttributes } from 'react'
+import React, { forwardRef } from 'react'
+
+import { Radio } from '../..'
+import { useRadioOption } from './useRadioOption'
+
+export interface RadioOptionProps
+  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'type'> {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+  /**
+   * Attribute to set aria-label option as required.
+   */
+  label: string
+  /**
+   * Value to be emitted when radio is clicked.
+   */
+  value: string | number
+}
+
+const RadioOption = forwardRef<HTMLInputElement, RadioOptionProps>(
+  function RadioOption(
+    { label, value, children, testId = 'store-radio-option', ...otherOptions },
+    ref
+  ) {
+    const { name, selectedValue, onChange } = useRadioOption()
+
+    return (
+      <label aria-label={label} data-store-radio-option>
```

Comment:
That was a discussion in [this comment](https://github.com/vtex/faststore/pull/1033#discussion_r756221033), the aria-label is important, because if the children are a complex type, accessibility is compromised. We ended up preferring the user to put the text as a child by themselves.